### PR TITLE
test(ivy): move property and attribute tests to acceptance

### PR DESCRIPTION
**Pull request by benlesh**
- splits existing property acceptance tests into property_binding and property_interpolation tests
- ports property and attribute related tests from render3 instructions tests to acceptance tests
- removes redundant or unnecessary tests that are covered by existing acceptance tests

Other things to note:

1. `ngDevMode` counter assertions were removed because they don't pass in ViewEngine. I recommend we create a separate battery of tests for those, because they were a little redundant through these tests in a lot of cases.
2. Not all tests could be moved. Some tests are really unit tests for the instruction and should stay.

### CODE REVIEW

File: packages/core/test/acceptance/attributes_spec.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      badUrl = 'javascript:true';
+    }
+
+    TestBed.configureTestingModule({declarations: [Comp]});
+    const fixture = TestBed.createComponent(Comp);
+    fixture.detectChanges();
+
+    const a = fixture.debugElement.query(By.css('a')).nativeElement;
+    // NOTE: different browsers will add `//` into the URI.
+    expect(a.href.indexOf('unsafe:')).toBe(0);
```
Comment (kara) [inline packages/core/test/acceptance/attributes_spec.ts]: Looks like we lost the second part of this test that checks `bypassSanitizationTrustUrl` works. Can we re-add it? e.g. something similar to [this](https://github.com/angular/angular/pull/30328/files#diff-0483f62e84f992bd169dee41626776e1R143)

File: packages/core/test/render3/instructions_spec.ts [hunk 2]
```diff
@@ -45,123 +44,6 @@ describe('instructions', () => {
...
-  describe('bind', () => {
-    it('should update bindings when value changes', () => {
-      const t = new TemplateFixture(createAnchor, () => {}, 1, 1);
-
-      t.update(() => ɵɵelementProperty(0, 'title', ɵɵbind('Hello')));
-      expect(t.html).toEqual('<a title="Hello"></a>');
-
-      t.update(() => ɵɵelementProperty(0, 'title', ɵɵbind('World')));
-      expect(t.html).toEqual('<a title="World"></a>');
-      expect(ngDevMode).toHaveProperties({
```
Comment (kara) [inline packages/core/test/render3/instructions_spec.ts]: Think we should also keep this basic devMode test to check properties are set the correct number of times when changing the value.

File: packages/core/test/render3/instructions_spec.ts [hunk 3]
```diff
@@ -45,123 +44,6 @@ describe('instructions', () => {
...
-      expect(t.html).toEqual('<a title="Hello"></a>');
-
-      t.update();
-      expect(t.html).toEqual('<a title="Hello"></a>');
-      expect(ngDevMode).toHaveProperties({
-        firstTemplatePass: 1,
-        tNode: 2,  // 1 for hostElement + 1 for the template under test
-        tView: 2,  // 1 for rootView + 1 for the template view
-        rendererCreateElement: 1,
-        rendererSetProperty: 1
```
Comment (kara) [inline packages/core/test/render3/instructions_spec.ts]: We also need to keep this devMode test (the point of it is to check that setProperty only gets called once if the value doesn't change and this is lost in the acceptance test)

File: packages/core/test/render3/instructions_spec.ts [hunk 4]
```diff
@@ -45,123 +44,6 @@ describe('instructions', () => {
...
-      t.update(
-          () => ɵɵelementAttribute(
-              0, 'title', bypassSanitizationTrustUrl('javascript:true'), ɵɵsanitizeUrl));
-      expect(t.html).toEqual('<div title="javascript:true"></div>');
-      expect(ngDevMode).toHaveProperties({
-        firstTemplatePass: 1,
-        tNode: 2,  // 1 for div, 1 for host element
-        tView: 2,  // 1 for rootView + 1 for the template view
-        rendererCreateElement: 1,
-        rendererSetAttribute: 2
```
Comment (kara) [inline packages/core/test/render3/instructions_spec.ts]: Also seems like we should keep a test to check `setAttribute` is set the correct number of times

File: packages/core/test/render3/instructions_spec.ts [hunk 5]
```diff
@@ -254,762 +94,6 @@ describe('instructions', () => {
...
-                'title',
-                ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '']);
-          });
-        }).not.toThrow();
-      });
-    });
-  });
-
-  describe('elementProperty', () => {
-    it('should use sanitizer function when available', () => {
```
Comment (kara) [inline packages/core/test/render3/instructions_spec.ts]: It looks like we don't have an equivalent test for this anymore (unless I'm just missing it?). Can you add one for properties that checks that the sanitizer can be used? I see one that checks unsafe properties are sanitized, but not one that uses the bypass feature.

File: packages/core/test/acceptance/property_binding_spec.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    TestBed.configureTestingModule({declarations: [App]});
+    const fixture = TestBed.createComponent(App);
+    fixture.detectChanges();
+    const img = fixture.nativeElement.querySelector('img');
+
+    expect(img.src.indexOf('unsafe:')).toBe(0);
+  });
+
+  it('should stringify non-string values', () => {
```
Comment (kara) [inline packages/core/test/acceptance/property_binding_spec.ts]:
```suggestion
  it('should not stringify non-string values', () => {
```

File: packages/core/test/render3/instructions_spec.ts [hunk 7]
```diff
@@ -254,762 +94,6 @@ describe('instructions', () => {
...
-          firstTemplatePass: 1,
-          tNode: 2,  // 1 for div, 1 for host element
-          tView: 2,  // 1 for rootView + 1 for the template view
-          rendererCreateElement: 1,
-          rendererSetProperty: 2,
-        });
-      });
-
-      it('should chain', () => {
-        // <div title="{{123}}" accesskey="{{'A'}}"></div>
```
Comment (kara) [inline packages/core/test/render3/instructions_spec.ts]: Can we add an acceptance test for this too? I agree we should keep one in instructions to test chaining works (saw your comment), but we should also have an integration version.

File: packages/core/test/acceptance/attributes_spec.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
+/**
```
Comment (kara) [inline packages/core/test/acceptance/attributes_spec.ts]: Thanks for breaking these out into their own file :)

**Review by kara — CHANGES_REQUESTED** [review]
Mostly looks good, main thing is that I think we should re-add some of the devMode counter tests

File: packages/core/test/render3/instructions_spec.ts (hunk 5 above)
Comment (benlesh) [reply packages/core/test/render3/instructions_spec.ts]: Ah... [this was pretty close](https://github.com/angular/angular/pull/30321/files#diff-4dd20352d46971d819ebc7600e80b26aR99), but it didn't test the bypass path. I'll add that.

**Review by kara — APPROVED** [review]
LGTM, thanks for the fixes!

**Pull request closed without merging**